vj4/ui/templates/components/problem.html

```diff
@@ -10,7 +10,7 @@
   {%- endif %}
   >
   {%- endif %}
-    {% if pdoc['doc_id']|string|length < 10 %}P{{ pdoc['doc_id'] }} {% endif %}{{ pdoc['title'] }}
+    {% if pdoc['doc_id'] is number %}P {{ pdoc['doc_id'] }}{% elif pdoc['doc_id']|string|length < 10 %} {{ pdoc['doc_id'] }} {% endif %}{{ pdoc['title'] }}
```

Comment:
The logic here seems complicated and incorrect. Can we implement them at Python side?

Reply:
Then we have to make a new template but I think it's not necessary.

Reply:
Actually the logic is:
If it is a number,we add a 'p' before.
Else if it's not an object id, show the specified string PID.
Otherwise, no ID will be shown.

Reply:
You added a space between P and the numeric `doc_id`.

I'm not sure what `number` is in jinja2 but I suspect it includes floating pointer number which should not be included.

Why is there a magic number `10`?

Again you added some random spaces in the string `doc_id` rendering.

Why is not displaying `doc_id` preferred in some situations?

Reply:
Oh..I added this space by mistake,I'll fix later.
If we always show doc_id,It will show pid 'A123' as 'PA12'

Reply:
Acturally I don't wrote the number 10.
It's in older version.

Reply:
Now it looks like this:
![image](https://user-images.githubusercontent.com/29992205/57569955-1b312000-742f-11e9-888a-936de6ff7a61.png)